### PR TITLE
Add and remove

### PR DESCRIPTION
**Pull request by J-C-S-V**
Here is what I did on this milestone:

- Remove all hardcoded items from the tasks array.
- Create a new JavaScript file for the new functionality.
- Add functions to add, edit and remove lists
- Set dynamic indexes for the objects

### CODE REVIEW

**Review by topeogunleye — COMMENTED** [review]
Hi @J-C-S-V, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project of Activity: To-Do-List that has the README file of `Hello Microverse`

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/addTasks.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  li.appendChild(removeButton);
+
+  // Add event listeners for remove and edit buttons
+  removeButton.addEventListener('click', removeTasks);
+  editTasks(li);
+
+  input.value = '';
+
+  // Add local storage
+  localStorage.setItem('tasks', JSON.stringify(tasks));
```
Comment (ShadyShawkat) [inline src/addTasks.js]: - [x] Good job updating the local storage whenever a task is created. However, the purpose of local storage is to preserve the data which should work both ways. In other words, whenever the page is loaded the data should be imported from the local storage and displayed for the user. Please do so.

File: src/editTasks.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export default function editTasks(li) {
+  const taskSpan = li.querySelector('.list__item-description');
+  taskSpan.addEventListener('click', () => {
+    const newContent = prompt('Edit task');
+    if (newContent === null || newContent === '') return;
+    taskSpan.textContent = newContent;
+  });
+}
```
Comment (ShadyShawkat) [inline src/editTasks.js]: - [x] Please make sure when the user edits a task, the change is then reflected in the local storage.

File: src/removeTasks.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export default function removeTasks(event) {
+  event.stopPropagation();
+  event.target.parentNode.remove();
+}
```
Comment (ShadyShawkat) [inline src/removeTasks.js]: - [x] Please make sure when the user deletes a task, the change is then reflected in the local storage. 
- [x] Kindly ensure that when a task is deleted, all remaining items' indexes are updated, so they represent the current list order and are unique. As per [this](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=Deleting%20a%20task%20should%20update%20all%20remaining%20items%27%20indexes%2C%20so%20they%20represent%20the%20current%20list%20order%20and%20are%20unique(i.e.%20if%20you%27re%20deleting%20the%20first%20task%20index%201%20from%20the%20list%2C%20the%20index%20of%20the%20next%20task(2)%20should%20set%20to%201)..) requirement.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @J-C-S-V, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/removeTasks.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+import { ul, tasks } from './index.js';
+
+export default function removeTasks(event) {
+  event.stopPropagation();
+  const li = event.target.parentNode;
+  const index = Array.prototype.indexOf.call(ul.children, li);
+  tasks.splice(index, 1);
+  li.remove();
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
```
Comment (AmaduKamara) [inline src/removeTasks.js]: - [x] Just as the previous reviewer pointed out, when a task is deleted, the remaining task indexes should get updated and be ordered from 1...3. Currently when I added 5 tasks and deleted tasks 3 and 4, the remaining tasks index does not get updated in an orderly manner as expected. Kindly work on that. 👍🏽
- [x] Let's try to not have Linter checks disabled in our codebase. It is there to help us write better clean and more professional code. Kindly fix the error 👍🏽

Screenshot 2023-03-31 at 12 42 12 AM

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @J-C-S-V,

Good job so far on fixing some of the changes highlighted by the previous reviewer 👍🏽
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍🏽
✅ Linter checks passed 👍🏽
✅ Add, edit, and remove features work 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [x] Kindly work on updating the remaining tasks indexes when one is deleted 👍🏽
- [x] Let's fix all Linter errors and not disable Linter checks 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/removeTasks.js (hunk 4 above)
Comment (J-C-S-V) [reply src/removeTasks.js]: Hello @AmaduKamara,

Could you please guide me through the solution?. I'm really lost on how to solve the index requirement :/

**Review by TadesseDev — APPROVED** [review]
### Hi @J-C-S-V 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**